### PR TITLE
Fixed Python makeTheCall dictionary syntax

### PR DESCRIPTION
**Pull request by josephaxisa**
### The Problem

Currently the Python generator defines dictionaries using the `dict()` function, e.g. `dict(k1 = "v1", k2 = "v2")`. 

This generates incorrect syntax when the key contains periods or other special characters like hyphens, as it requires it to be quoted, e.g. `dict("foo.bar" = "v1")`. Additionally, passing keyword arguments into `dict()` is risky if the keys are not valid Python identifiers, e.g. `import`. 

### Solution
Initialize dictionaries using curly braces as it works for all kinds of keys. 
- Added `hashKeyQuote` and `hashSetSep` attributes to `ICodeGen`
- Removed overriding `argSet` method in PythonGen

Note that currently we do not capture the key types and all keys are quoted. This is a bug that is unlikely to be experienced but might need fixing later on. See [here](https://github.com/looker-open-source/sdk-codegen/blob/e00f8f7d2d6d138232ed21be231788c7898b9184/packages/sdk-codegen/src/python.gen.spec.ts#L1028-L1032).

### CODE REVIEW

File: packages/sdk-codegen/src/python.gen.spec.ts [hunk 1]
```diff
@@ -1025,33 +1025,43 @@ class MergeFields(model.Model):
         const oneItem = [1]
         const threeItems = ['Abe', 'Zeb', token]
         const inputs: IDictionary<any> = {
+          /**
+           * The below key is quoted in the generated dictionary. This is a bug
+           * that might need addressing later on.
+           */
```
Comment (jkaster) [inline packages/sdk-codegen/src/python.gen.spec.ts:1031]: This can wait for another PR, but can you explain what's going on here?

**Review by jkaster — DISMISSED** [review]
Good for this PR. Just one question for follow-up if it's not a quick fix.

File: packages/sdk-codegen/src/python.gen.spec.ts (hunk 1 above)
Comment (josephaxisa) [reply packages/sdk-codegen/src/python.gen.spec.ts:1031]: In Python it is possible to have an integer as a dictionary key. Consider
```python
x = { 1: "one", "2": "two" }
```
These two values would be accessed with `x[1]` and `x["2"]` respectively in Python. However, in the Python code generator we are quoting all dictionary keys.

**Comment by joeldodge79** [comment]
hrm, looks like we're missing CI for the paths touched in this PR. I'm guessing a separate CI file for now since the tssdk-ci only runs sdk tests? https://github.com/looker-open-source/sdk-codegen/blob/3c74927b0cd9409516ec027536dfeadfcfb65dfc/.github/workflows/tssdk-ci.yml#L93

File: packages/sdk-codegen/src/codeGen.ts [hunk 2]
```diff
@@ -652,6 +659,7 @@ export abstract class CodeGen implements ICodeGen {
   arrayClose = ']'
   hashOpen = '{'
   hashClose = '}'
+  hashKeyQuote = ''
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/codeGen.ts:662]: is this mean to be an empty string? or did you mean `'"'` or conversely `"'"`?

File: packages/sdk-codegen/src/python.gen.ts [hunk 3]
```diff
@@ -47,15 +47,17 @@ export class PythonGen extends CodeGen {
   indentStr = '    '
   argDelimiter = `,\n${this.indentStr.repeat(3)}`
   argSetSep = '='
+  hashSetSep = ': '
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/python.gen.ts:50]: same value as parent class

```suggestion
```

File: packages/sdk-codegen/src/python.gen.ts [hunk 4]
```diff
@@ -47,15 +47,17 @@ export class PythonGen extends CodeGen {
...
   propDelimiter = '\n'
   dataStructureDelimiter = ', '
   enumDelimiter = '\n'
   codeQuote = '"'
   typeOpen = '('
   typeClose = ')'
-  hashOpen = 'dict('
-  hashClose = ')'
+  hashOpen = '{'
+  hashClose = '}'
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/python.gen.ts:59]: ditto
```suggestion
```

**Review by joeldodge79 — CHANGES_REQUESTED** [review]
nice work @josephaxisa ! couple of small nits

File: packages/sdk-codegen/src/codeGen.ts (hunk 2 above)
Comment (joeldodge79) [reply packages/sdk-codegen/src/codeGen.ts:662]: oh, nm - javascript objects aren't quoted

Comment (jkaster) [reply packages/sdk-codegen/src/codeGen.ts:662]: Should be empty string in the base class

**Review by jkaster — APPROVED** [review]
Looks good

**Review by joeldodge79 — APPROVED** [review]
👍

**Pull request merged**